js/ui/iconGridLayout.js

```diff
@@ -379,6 +379,77 @@ const IconGridLayout = new Lang.Class({
                 Lang.bind(this, this._enumerateDirectoryFiles));
     },
 
+    _createFolderFile: function(name) {
+        let keyFile = new GLib.KeyFile();
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_NAME, name);
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_TYPE, GLib.KEY_FILE_DESKTOP_TYPE_DIRECTORY);
+
+        let dir = Gio.File.new_for_path(GLib.get_user_data_dir() + '/desktop-directories');
+        if (!dir.query_exists(null))
```

Comment:
I usually try to catch exceptions only when something is not expected to fail normally. This directory will exist right after the first folder is added to the desktop, so its existence is what's expected. But I will modify it as you say.